wp-includes/class-wp-service-workers.php

```diff
@@ -117,6 +117,31 @@ public function init() {
 			array( 'workbox-sw' )
 		);
 
+		$offline_page_id = (int) get_option( WP_Offline_Page::OPTION_NAME, 0 );
+		if ( $offline_page_id ) {
+
+			// Cache relevant theme assets.
+			$this->register_cached_route( '/wp-content/.*\.(?:css|gif|png|jpg|jpeg)', self::STRATEGY_STALE_WHILE_REVALIDATE );
+			$this->register(
+				'offline-sw',
+				array( $this, 'configure_offline_page' ),
+				'workbox-sw'
+			);
+
+			// Register precaching for offline page route.
+			$offline_post         = get_post( $offline_page_id );
+			$content_asset_routes = $this->get_offline_post_image_routes( $offline_post );
+			$this->register_precached_routes( array_merge(
+				array(
+					array(
+						'url'      => $this->remove_url_scheme( get_the_permalink( $offline_page_id ) ),
```

Comment:
I think actually we won't need `remove_url_scheme` because all of the URLs should be getting served as HTTPS anyway.